src/report/report_utils.py

```diff
@@ -0,0 +1,498 @@
+        for cell in row.cells:
+            cell.paragraphs[0].paragraph_format.alignment = text.WD_ALIGN_PARAGRAPH.CENTER
+            cell.vertical_alignment = table.WD_CELL_VERTICAL_ALIGNMENT.CENTER
+    return t
+
+
+def delete_paragraph(paragraph: Paragraph) -> None:
+    """
+    Deletes passed paragraph from parent element.
+
+    Args:
+        paragraph:
+            Paragraph to delete.
+
+    Returns:
+        None
+    """
+    p = paragraph._element
+    p.getparent().remove(p)
+    p._p = p._element = None
+
+
+def add_image_to_cell(cell: _Cell, image: str) -> None:
+    """
+    Adds image to provided table cell.
+
+    Args:
+        cell:
+            Reference to cell.
+        image:
+            File path to image.
+
+    Returns:
+        None
+    """
+    # image width must be smaller than cell width to accomodate pdf conversion inacurracies
+    cell.paragraphs[0].add_run().add_picture(image, Emu(cell.width * 0.85))
+
+
+def change_cells_text_bold(cells: [_Cell], value: bool) -> None:
+    """
+    Changes text bold setting for each cell in list.
+
+    Args:
+        cells:
+            List of cells to change text of.
+        value:
+            Bold value to set text to.
+
+    Returns:
+        None
+    """
+    for cell in cells:
+        t = cell.paragraphs[0].text
+        cell.paragraphs[0].text = f''
+        cell.paragraphs[0].add_run(t).bold = value
+
+
+def change_cells_text_color(cells: [_Cell], color: RGBColor) -> None:
+    """
+    Changes text color setting for each cell in list.
+
+    Args:
+        cells:
+            List of cells to change text of.
+        color:
+            RGB color to set text to.
+
+    Returns:
+        None
+    """
+    for cell in cells:
+        t = cell.paragraphs[0].text
+        cell.paragraphs[0].text = f''
+        cell.paragraphs[0].add_run(t).font.color.rgb = color
+
+
+def change_cells_background_color(cells: [_Cell], color: str) -> None:
+    """
+    Changes background color setting for each cell in list.
+
+    Args:
+        cells:
+            List of cells to change background of.
+        color:
+            RGB color to set background to.
+
+    Returns:
+        None
+    """
+    for cell in cells:
+        properties = cell._element.tcPr
+        try:
+            shading = properties.xpath('w:shd')[0]
+        except IndexError:
+            shading = OxmlElement('w:shd')
+        shading.set(qn('w:fill'), color)
+        properties.append(shading)
+
+
+def convert_docx_to_pdf(docx_path: str, pdf_directory: str) -> str:
+    """
+    Calls LibreOffice to convert provided docx_file to pdf.
+    Converted pdf file is saved in pdf_directory.
+    LibreOffice soffice utility is provided via environmental variable LIBREOFFICE.
+
+    Args:
+        docx_path:
+            Path to docx file to convert.
+        pdf_directory:
+            Directory to save pdf file in.
+
+    Returns:
+        Path to pdf file or empty if LIBREOFFICE is not set.
```

Comment:
I'd suggest just raising an exception in case of failure. Empty string has to be handled in a special way anyway, and an exception would serve the purpose better imo.